lib/assignDisabledRanges.mjs

```diff
@@ -61,101 +73,162 @@ export default function assignDisabledRanges(root, result) {
 
 	const configurationComment = result.stylelint.config?.configurationComment;
 
-	root.walkComments((comment) => {
-		if (inlineEnd) {
-			// Ignore comments already processed by grouping with a previous one.
-			if (inlineEnd === comment) inlineEnd = null;
+	root.walk((node) => {
+		if (isComment(node)) {
+			if (inlineEnd) {
+				// Ignore comments already processed by grouping with a previous one.
+				if (inlineEnd === node) inlineEnd = null;
 
-			return;
-		}
+				return;
+			}
 
-		const nextComment = comment.next();
+			const nextComment = node.next();
+
+			// If any of these conditions are not met, do not merge comments.
+			if (
+				!(
+					!isStandardSyntaxComment(node) &&
+					isConfigurationComment(node.text, configurationComment) &&
+					nextComment &&
+					isComment(nextComment) &&
+					(node.text.includes('--') || nextComment.text.startsWith('--'))
+				)
+			) {
+				checkComment(node, node.source, node.text);
+
+				return;
+			}
 
-		// If any of these conditions are not met, do not merge comments.
-		if (
-			!(
-				!isStandardSyntaxComment(comment) &&
-				isConfigurationComment(comment, configurationComment) &&
-				nextComment &&
-				nextComment.type === 'comment' &&
-				(comment.text.includes('--') || nextComment.text.startsWith('--'))
-			)
-		) {
-			checkComment(comment);
+			let lastLine = (node.source && node.source.end && node.source.end.line) || 0;
+			const fullComment = node.clone();
 
-			return;
+			let current = nextComment;
+
+			while (
+				!isStandardSyntaxComment(current) &&
+				!isConfigurationComment(current.text, configurationComment)
+			) {
+				const currentLine = (current.source && current.source.end && current.source.end.line) || 0;
+
+				if (lastLine + 1 !== currentLine) break;
+
+				fullComment.text += `\n${current.text}`;
+
+				if (fullComment.source && current.source) {
+					fullComment.source.end = current.source.end;
+				}
+
+				inlineEnd = current;
+				const next = current.next();
+
+				if (!next || !isComment(next)) break;
+
+				current = next;
+				lastLine = currentLine;
+			}
+
+			checkComment(fullComment, fullComment.source, fullComment.text);
 		}
 
-		let lastLine = (comment.source && comment.source.end && comment.source.end.line) || 0;
-		const fullComment = comment.clone();
+		if (isRule(node)) {
+			let offset = 0;
+			const selector = node.raws?.selector?.raw;
 
-		let current = nextComment;
+			if (selector && containsComments(selector)) {
+				checkCommentContainingNode(node, selector, offset);
+			}
```

Comment:
[suggest] This combination of `containsComments` and `checkCommentContainingNode` often appears, so I feel we can refactor like this: 👇🏼 

```suggestion
			checkCommentContainingNode(node, selector, offset);
```

Inlining `containsComments()`, e.g.

```js
/**
 * @param {PostcssNode} node
 * @param {string | undefined} part
 * @param {number} offset
 */
function checkCommentContainingNode(node, part, offset) {
	if (!(part && part.includes('/*') && part.includes('*/'))) return;

	// ...
}
```

Reply:
Good call, thank you